File: Agile Development Practices/Code/DotNet/videoworld/VideoWorld/Models/NewReleasePrice.cs
namespace VideoWorld.Models
{
    public class NewReleasePrice : IPrice
    {
        public decimal GetCharge(int periodInDays)
        {
            return periodInDays*3.00m;
        }

        public int GetFrequentRenterPoints(int daysRented)
        {
            // add bonus for a two day new release rental
            if (daysRented > 1)
                return 2;
            else
                return 1;
        }
    }
}

File: Agile Development Practices/Code/Java/videoworld/src/main/java/com/thoughtworks/videorental/domain/NewReleasePrice.java
package com.thoughtworks.videorental.domain;

public class NewReleasePrice implements Price {

	public double getCharge(final int daysRented) {
		return daysRented * 3;
	}

	public int getFrequentRenterPoints(final int daysRented) {
		// add bonus for a two day new release rental
		if (daysRented > 1)
			return 2;
		else
			return 1;
	}

}
